examples/sift.rs

```diff
@@ -0,0 +1,68 @@
+extern crate cv;
+#[macro_use]
+extern crate itertools;
+
+use cv::features2d::*;
+use cv::highgui::*;
+use cv::imgcodecs::ImageReadMode;
+use cv::imgproc::*;
+use cv::*;
+use std::iter::once;
+use std::os::raw::c_int;
+
+fn main() {
+    let args: Vec<_> = std::env::args().collect();
+    if args.len() != 2 {
+        println!("Usage: sift ImageToLoadAndDisplay");
+        std::process::exit(-1);
+    }
+
+    let mat = Mat::from_path(&args[1], ImageReadMode::Color).expect("Failed to read from path");
+
+    if !mat.is_valid() {
+        println!("Could not open or find the image");
+        std::process::exit(-1);
+    }
+
+    let blue = mat.mix_channels(1, 1, [(0, 0)]);
+    let green = mat.mix_channels(1, 1, [(1, 0)]);
+    let red = mat.mix_channels(1, 1, [(2, 0)]);
+
+    let mask = Mat::new();
+    let sift: SIFT = SIFTBuilder::default().into();
+    let (red_keypoints, _) = sift.detect_and_compute(&red, &mask);
+    let (green_keypoints, _) = sift.detect_and_compute(&green, &mask);
+    let (blue_keypoints, _) = sift.detect_and_compute(&blue, &mask);
+
+    for (kps, (c1, c2)) in
+        [red_keypoints, green_keypoints, blue_keypoints]
+            .iter()
+            .zip(&[(&green, &blue), (&blue, &red), (&green, &red)])
+    {
+        for kp in kps {
+            let x = kp.point.x as c_int;
```

Comment:
use regular `i32` instead of all `c_int`. See http://answers.opencv.org/question/184248/does-opencv-runs-on-platforms-with-different-int-size/ . I believe I had a gurard that checks that `c_int` is actually 32bit integer.